samples/shopping-cart-addview/src/main/java/shoppingcart/application/ShoppingCartView.java

```diff
@@ -0,0 +1,82 @@
+package shoppingcart.application;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import akka.javasdk.annotations.ComponentId;
+import akka.javasdk.annotations.Consume;
+import akka.javasdk.annotations.Query;
+import akka.javasdk.view.TableUpdater;
+import akka.javasdk.view.View;
+import shoppingcart.domain.ShoppingCartEvent;
+
+@ComponentId("shopping_cart_view")
+public class ShoppingCartView extends View {
+  @Query("SELECT * FROM shopping_cart_view WHERE cartId = :cartId")
```

Comment:
Doesn't make much sense (to me) to have a view query on the entity id. That can be served in a more efficient way by the entity directly. Can't we come up with something more realistic, such as
* carts that are checked out between a date interval
* carts that contain a certain product
* carts with more than 10 products

Reply:
🤔 my main goal with this sample was to show segregating the different responsibilities of the models and to show a view that (eventually) has more information than the entity. This was to make way for when, in the next step in the journey, we add AI-supplied recommendations which will appear in the view and not the entity

Reply:
How would we add additional information to the view? Through the events?

No sure if aligned with the original goal, but here is some alternative idea. 

* a shopping cart is associated with a user ID
* we have a consumer that listen to shopping cart events 
* the consumer users some AI assistance to build up a UserProfile (can be a KVE)
* the user profile is used to build the product recommendations using information from different checkouts

As such, the recommendation is not based on the previous shopping activity, but it's an aggregation of all shopping behavior of a given user.

Reply:
Yes, the information is on the events (see the latest commit). The entity only needs the product ID and quantity, while the view contains the description and name. This phase of the shopping cart tour doesn't utilize AI in any form. Adding the view is a between step between the first quickstart and adding AI